### PR TITLE
[feature] Refactor of ExtraTrees preview for eventual merge with Random Forest

### PR DESCRIPTION
**Pull request by icfaust**
# Description
This generalizes the ExtraTrees code to allow for commonality between Random Forest and Extra Trees for the eventual move out of preview.  It now makes the dependency on scikit-learn on a class above the public API.

Changes proposed in this pull request:
- scikit-learn ExtraForestRegressor inheritance to ForestRegressor inheritance
- scikit-learn ExtraForestClassifier inheritance to ForestClassifier inheritance

### CODE REVIEW

File: sklearnex/preview/ensemble/extra_trees.py [hunk 1]
```diff
@@ -208,150 +212,55 @@ def estimators_(self, estimators):
...
-            self.ccp_alpha = ccp_alpha
-            self.max_samples = max_samples
-            self.max_bins = max_bins
-            self.min_bin_size = min_bin_size
+        # The splitter is recognized here for proper dispatching.
+        self._estimator = estimator  # TODO: Verify if this is done in older verions
+        if self._estimator.__class__ == DecisionTreeClassifier:
+            self._onedal_classifier = onedal_RandomForestClassifier
+        elif self._estimator.__class__ == ExtraTreeClassifier:
+            self._onedal_classifier = onedal_ExtraTreesClassifier
```
Comment (KulikovNikita) [inline sklearnex/preview/ensemble/extra_trees.py:255]: Please also add `else` statement

**Review by KulikovNikita — COMMENTED** [review]
Looks fine to me

**Comment by Alexsandruss** [comment]
@KulikovNikita, @icfaust, where is green PreCommit job for last commit?

**Comment by icfaust** [comment]
http://intel-ci.intel.com/ee4b2c8e-e60f-f127-9bc2-a4bf010d0e2e

**Comment by samir-nasibli** [comment]
It is expected that #1424 will be merged before. And also I don't understand why we are merging PR with red CI?

**Comment by samir-nasibli** [comment]
@icfaust @KulikovNikita Please, give more time to reviewers for the review. Whole the time this PR was marked as a draft

**Comment by samir-nasibli** [comment]
> # Description
> This generalizes the ExtraTrees code to allow for commonality between Random Forest and Extra Trees for the eventual move out of preview. It now makes the dependency on scikit-learn on a class above the public API.
> 
> Changes proposed in this pull request:
> 
> * scikit-learn ExtraForestRegressor inheritance to ForestRegressor inheritance
> * scikit-learn ExtraForestClassifier inheritance to ForestClassifier inheritance

> This generalizes the ExtraTrees code to allow for commonality between Random Forest and Extra Trees for the eventual move out of preview. 

this is not the reason why it is on preview. The reason is perf issues and disabled tests that should be resolved.

**Comment by icfaust** [comment]
@samir-nasibli , if there are additional rules to how we review code for this codebase, it should be written down and documented. These changes follow through on what was presented to the architecture board for what RF/ET should look like out of preview.  I assumed everything was good to go with a positive code-owner review and CI check. The private CI issues were described as related to linalg issues, otherwise green, and the public CI passes all tests. The additional public nightly CI check on this codebase being red (issues on the codebase associated with anaconda) is not due to extratrees algorithms.

If this is not our procedure for PRs than please let me and others know. For example, if one code owner isn't sufficient, lets try and make it two for all PRs and have a discussion about it. Otherwise, it makes @KulikovNikita 's review meaningless.

**Comment by samir-nasibli** [comment]
@icfaust Thank you very much for your work, actually it looks very good to me.
I am deeply sorry for misunderstanding. My feedbacks was based only on https://github.com/intel/scikit-learn-intelex/pull/1425#issuecomment-1705631411 check. So they could be ignored.

About the PR rules/templates, as an option we can continue discussion here https://github.com/intel/scikit-learn-intelex/pull/1388. Feel free to share your thoughts

**Pull request merged**